### PR TITLE
editoast: migrate endpoint GET rolling_stock/{id}/livery/{id}

### LINKED ISSUE #3300: migrate GET /rolling-stock/{id}/livery endpoint to editoast
# Description
Migrate `GET /rolling-stock/{id}/livery` endpoint to editoast. This endpoint returns the compound image of the requested livery.

# Goal
Drop API service.

# Acceptance Criteria
- [ ] the frontend uses the editoast endpoint and the user does not see any difference
- [ ] the endpoint doesn't exist anymore in API

### PR DESCRIPTION
**Pull request by clarani**
close #3300 

- migrate endpoint GET `rolling_stock/{id}/livery/{id}` to rust
- remove delete on cascade on rolling_stock_compound_image (otherwise, when we delete the compound image, it automatically deletes the livery too, which is not the expected behaviour)
- add tests & documentation

Later, we should change the database structure to use the delete on cascade (which implies removing the compound_image_id of the livery and add the livery_id in the compound_image = reverse the relationship)

### CODE REVIEW

**Review by flomonster — APPROVED** [review]
Good for me but this code will be highly refactored.
We should store liveries images in the document table.
We will use the new model system to interact with the DB.

**Review by younesschrifi — APPROVED** [review]

File: editoast/src/schema/rolling_stock_image.rs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+                Ok(image_id) => Ok(image_id),
+                Err(e) => Err(e.into()),
+            }
+        })
+        .await
+        .unwrap()
+    }
+
+    /// Delete a compound image and set the compound_image_id to null in its livery
+    pub async fn delete(db_pool: Data<DbPool>, livery_id: i64, image_id: i64) -> Result<()> {
```
Comment (Castavo) [inline editoast/src/schema/rolling_stock_image.rs:115]: You use `livery_id` only for the error message
Furthermore you say "set the compound_image_id to null" is it done automatically ?

Comment (Castavo) [reply editoast/src/schema/rolling_stock_image.rs:115]: Oh okay it's done automatically thanks to Django, I find the function's docstring a bit confusing then

**Comment by Castavo** [comment]
Nice PR !
I've learnt about `Selectable` today :grin:

**Review by Castavo — APPROVED** [review]
Nice PR !
I've learnt about `Selectable` today :grin:

**Review by anisometropie — APPROVED** [review]

**Pull request merged**